MakerTracker/ClientApp/src/app/Profiles/init-profile/init-profile.component.html

```diff
@@ -1,290 +1,288 @@
+            >
+              <mat-optgroup *ngFor="let group of products" [label]="group.name" [hidden]="!group.products.length">
+                <mat-option *ngFor="let product of group.products" [value]="product">
+                  {{ product.name }}
+                </mat-option>
+              </mat-optgroup>
+            </mat-select>
+          </mat-form-field>
+          <div formArrayName="needs">
+            <ng-container *ngFor="let product of requestorFormGroup.get('needs').controls; let i = index">
+              <div [formGroupName]="i" class="maker-stock flex-row stackable flex-between align-center">
+                <span class="product-label">{{ productMap.get(product.value.productId) }}</span>
+                <mat-form-field>
+                  <input
+                    formControlName="quantity"
+                    matInput
+                    type="number"
+                    required
+                    placeholder="Quantity"
+                    min="0"
+                    step="1"
+                  />
+                </mat-form-field>
+                <mat-form-field>
+                  <mat-label>Needed by</mat-label>
+                  <input matInput [matDatepicker]="picker" [min]="today" formControlName="dueDate" />
+                  <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
+                  <mat-datepicker #picker [startAt]="today"></mat-datepicker>
+                  <mat-hint>optional</mat-hint>
+                </mat-form-field>
+                <mat-form-field>
+                  <mat-label>Special Instructions</mat-label>
+                  <textarea matInput formControlName="specialInstructions" rows="4"></textarea>
+                  <mat-hint>optional</mat-hint>
+                </mat-form-field>
+              </div>
+            </ng-container>
+          </div>
+        </div>
+        <div class="flex-row flex-between">
+          <button mat-button matStepperPrevious mat-raised-button color="primary">
+            <mat-icon>arrow_back</mat-icon> Back
+          </button>
+          <button mat-button matStepperNext mat-raised-button color="primary">
+            Next<span *ngIf="!requestorFormGroup.get('needs').length"> (Skip)</span> <mat-icon>arrow_forward</mat-icon>
+          </button>
+        </div>
+      </form>
+    </mat-step>
+    <mat-step>
+      <ng-template matStepLabel>Done!</ng-template>
+      <p>That's all we need!</p>
+      <p>You can explore more by proceeding to your dashboard.</p>
       <div class="flex-row flex-between">
-        <button mat-button matStepperPrevious mat-raised-button color="primary">
-          <mat-icon>arrow_back</mat-icon> Back
-        </button>
-        <button mat-button matStepperNext mat-raised-button color="primary">
-          Next<span *ngIf="!requestorFormGroup.get('needs').length"> (Skip)</span> <mat-icon>arrow_forward</mat-icon>
+        <button mat-button matStepperPrevious>Back</button>
```

Comment:
Added spinner and isBusy check here to set `disabled` and swapped the icon with progress spinner.